firefox-mobile/addon.js

```diff
@@ -0,0 +1,19 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import initAddon from "../shared/base.js";
+
+const REPORTER_ID = "addon-reporter-firefox-mobile";
+
+browser.runtime.getBrowserInfo().then(function({ version }) {
+  // Proceed if we're not running on Nightly.
+  if (!version.includes("a")) {
+    // Fennec doesn't support context menu APIs
+    initAddon(REPORTER_ID, { createContextMenu: false });
+  }
+
+  // Uninstall, without prompts, because this addon makes no sense on Nightly
+  // (it's built-in by default)
+  browser.management.uninstallSelf();
```

Comment:
This would cause the addon to simply uninstall itself, which is probably *not quite* desired here, so the call should be in the else clause.

Actually, this shouldn't be in the else clause, since we'd then have a negated-if with an else. 

```js
if (version.includes("a")) {
  // Uninstall, without prompts, because this addon makes no sense on Nightly
  // (it's built-in by default)
  browser.management.uninstallSelf();
} else {
  // Fennec doesn't support context menu APIs
  initAddon(REPORTER_ID, { createContextMenu: false });
}
```

Is probably the nicest way. Or an early return inside the if.

Reply:
Great catch, thanks. :)